examples/fix_invalid_speakers.py

```diff
@@ -0,0 +1,386 @@
+                if (clean_reason := reason.strip())
+            ),
+            corrected_email=csv_row['corrected_email'].strip(),
+            corrected_id=csv_row['corrected_id'].strip(),
+            corrected_name=csv_row['corrected_name'].strip(),
+        )
+
+    def to_csv_row(self) -> dict[str, Optional[str]]:
+        return {
+            'email': self.email,
+            'ids': ', '.join(sorted(self.ids)),
+            'names': ', '.join(sorted(self.names)),
+            'url': self.url,
+            'reasons': ', '.join(sorted(self.reasons)),
+            'corrected_email': self.corrected_email,
+            'corrected_id': self.corrected_id,
+            'corrected_name': self.corrected_name,
+        }
+
+
+def _get_speakers_by_oid(msc: MediaServerClient) -> dict[str, list[Speaker]]:
+    speakers = {}
+    catalog = msc.api('catalog/get-all/', params={'format': 'flat', 'timings': 'yes'})
+    for key in ['channels', 'videos', 'lives', 'photos']:
+        for obj in catalog.get(key, []):
+            oid = obj['oid']
+            if key == 'channels':
+                obj = obj.get('default_settings')
+                if not obj:
+                    continue
+            speakers_ids, speakers_emails, speakers_names = [], [], []
+            if obj.get('speaker_id'):
+                speakers_ids = [
+                    speaker_id.strip()
+                    for speaker_id in obj.get('speaker_id').split('|')
+                ]
+            if obj.get('speaker_email'):
+                speakers_emails = [
+                    speaker_email.strip()
+                    for speaker_email in obj.get('speaker_email').split('|')
+                ]
+            if obj.get('speaker'):
+                speakers_names = [
+                    speaker_name.strip()
+                    for speaker_name in obj.get('speaker').split('|')
+                ]
+            if len({
+                len(lst)
+                for lst in [speakers_emails, speakers_ids, speakers_names]
+                if lst
+            }) > 1:
+                logger.error(
+                    f'Media "{oid}" will be ignored because its speakers '
+                    f'cannot be parsed reliably: {[speakers_emails, speakers_ids, speakers_names]}'
+                )
+                continue
+            speakers[oid] = [
+                Speaker(speaker_email, speaker_id, speaker_name)
+                for speaker_email, speaker_id, speaker_name in zip_longest(
+                    speakers_emails, speakers_ids, speakers_names
+                )
+            ]
+    return speakers
+
+
+def _get_users(msc: MediaServerClient, page_size=500):
+    users = []
+    offset = 0
+    response = msc.api('users/', params={'limit': page_size, 'offset': offset})
+    while response['users']:
+        users += response['users']
+        offset += page_size
+        response = msc.api('users/', params={'limit': page_size, 'offset': offset})
```

Comment:
If the length of the last `response['users']` is less than `page_size`, there is no need to make further requests.